### ISSUE TITLE
[BUG]plugin can't loaded in spigot-fork

### ISSUE THREAD

**Issue by Frodwo** [Ghost-chu/QuickShop-Reremake#931]
[19:01:28] [Server thread/ERROR]: Could not load 'plugins\QuickShop-4.0.7.11.jar' in folder 'plugins'
org.bukkit.plugin.InvalidPluginException: java.lang.SecurityException: SHA-256 digest error for org/maxgamer/quickshop/QuickShop.class
	at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:141) ~[patched_1.16.5.jar:git-Purpur-1079]
	at org.bukkit.plugin.SimplePluginManager.loadPlugin(SimplePluginManager.java:397) ~[patched_1.16.5.jar:git-Purpur-1079]
	at org.bukkit.plugin.SimplePluginManager.loadPlugins(SimplePluginManager.java:305) ~[patched_1.16.5.jar:git-Purpur-1079]
	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.loadPlugins(CraftServer.java:389) ~[patched_1.16.5.jar:git-Purpur-1079]
	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:264) ~[patched_1.16.5.jar:git-Purpur-1079]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1070) ~[patched_1.16.5.jar:git-Purpur-1079]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Purpur-1079]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_291]
Caused by: java.lang.SecurityException: SHA-256 digest error for org/maxgamer/quickshop/QuickShop.class
	at sun.security.util.ManifestEntryVerifier.verify(Unknown Source) ~[?:1.8.0_291]
	at java.util.jar.JarVerifier.processEntry(Unknown Source) ~[?:1.8.0_291]
	at java.util.jar.JarVerifier.update(Unknown Source) ~[?:1.8.0_291]
	at java.util.jar.JarVerifier$VerifierStream.read(Unknown Source) ~[?:1.8.0_291]
	at java.io.InputStream.read(Unknown Source) ~[?:1.8.0_291]
	at com.google.common.io.ByteStreams.copy(ByteStreams.java:106) ~[patched_1.16.5.jar:git-Purpur-1079]
	at com.google.common.io.ByteStreams.toByteArray(ByteStreams.java:166) ~[patched_1.16.5.jar:git-Purpur-1079]
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:151) ~[patched_1.16.5.jar:git-Purpur-1079]
	at org.bukkit.plugin.java.JavaPluginLoader.getClassByName(JavaPluginLoader.java:216) ~[patched_1.16.5.jar:git-Purpur-1079]
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:123) ~[patched_1.16.5.jar:git-Purpur-1079]
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:111) ~[patched_1.16.5.jar:git-Purpur-1079]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_291]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_291]
	at java.lang.Class.forName0(Native Method) ~[?:1.8.0_291]
	at java.lang.Class.forName(Unknown Source) ~[?:1.8.0_291]
	at org.bukkit.plugin.java.PluginClassLoader.<init>(PluginClassLoader.java:79) ~[patched_1.16.5.jar:git-Purpur-1079]
	at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:137) ~[patched_1.16.5.jar:git-Purpur-1079]
	... 7 more

**Comment by Ghost-chu** [comment]
Attention - High security risk about unofficial builds or infected servers:

When your server trying loading infected plugins, the virus will inject bad codes to your server core and maybe have ability to infect your other regluar jar files, not only it will hide itself but also it will open the backdoors to virus purchaser of full control on your server, so it even can hack in your host system with other hack tools that remotely installed on your Minecraft server.

Please pay a attention to that, make sure your're download server core or plugins from official site and check the file hash if plugin author provided it.

For QuickShop, we only publish our plugins on:
SpigotMC: https://www.spigotmc.org/resources/quickshop-reremake-1-16-ready-multi-currency.62575/
CraftBukkit Devs: https://dev.bukkit.org/projects/320536
CodeMC: https://ci.codemc.io/job/Ghost-chu/view/QuickShop/

NEVER TRUST JARS THAT OTHERS SEND TO YOU

QuickShop jars contains build data that can allow you check if jar modified. After new build, we also added built-in self checker and you will receive the warnings.
The warning usually looks like SHA-1 digest error or Security risk detected and most of them point to org/maxgamer/quickshop/QuickShop.class. Once you seen one of them, that mean you server has been infected.

If you confirm infected, delete it immediately, make a full backup of you server files to prevent hacking and also check your server core jar to make sure it not infected.

**Comment by sandtechnology** [comment]
QuickShop jars contains build data that can allow you check if jar modified. After new build, we also added built-in self checker and you will receive the warnings.
The warning usually looks like SHA-1 digest error or Security risk detected and most of them point to org/maxgamer/quickshop/QuickShop.class. Once you seen one of them, that mean you server has been infected.

If you confirm infected, delete it immediately, make a full backup of you server files to prevent hacking and also check your server core jar to make sure it not infected.

I recommended to use this anti-malware tool to scan the server: https://www.spigotmc.org/resources/spigot-anti-malware-detects-over-300-malicious-plugins.64982/

Usually is caused by the "L10" malware
